spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java

```diff
@@ -25,35 +25,33 @@
 
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.AutoConfigureBefore;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
 import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
-import org.springframework.cloud.gcp.autoconfigure.core.AppEngineCondition;
 import org.springframework.cloud.gcp.autoconfigure.core.GcpContextAutoConfiguration;
 import org.springframework.cloud.gcp.autoconfigure.core.GcpProperties;
 import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
 import org.springframework.context.annotation.Primary;
 import org.springframework.util.StringUtils;
 
 /**
- * Google Cloud SQL starter.
- *
- * <p>Provides Google Cloud SQL instance connectivity through Spring JDBC by providing only a database and instance
- * connection name.
+ * Provides Google Cloud SQL instance connectivity through Spring JDBC by providing only a
+ * database and instance connection name.
  *
  * @author João André Martins
  */
 @Configuration
 @EnableConfigurationProperties({ GcpCloudSqlProperties.class, DataSourceProperties.class })
-@ConditionalOnClass(com.google.cloud.sql.mysql.SocketFactory.class)
```

Comment:
Since it's no longer conditional on `com.google.cloud.sql.mysql.SocketFactory`, will it always be turned on regardless of the classpath?

Reply:
Ha! Good catch! I think should be `@ConditionalOnBean(CloudSqlJdbcInfoProvider.class)`

Reply:
That's actually a really good point.
I think we will now need the SQL flag to always be turned on for this one as well.
I don't think we can rely on the classpath anymore...

Reply:
How cool it would be to have an OR `@ConditionalOnClass` here :smile:

Reply:
@artembilan I'm wondering at the time when `@ConditionalOnClass` is evaluated, would `@Import` have already kicked in?

Reply:
Can we just split this class (`GcpCloudSqlAutoConfiguration`) into the `MySqlJdbcInfoProviderAutoConfiguration` and `PostgreSqlJdbcInfoProviderAutoConfiguration`?

Reply:
Right, `@Import` are loaded first: 
```
// Process any @Import annotations
processImports(configClass, sourceClass, getImports(sourceClass), true);

// Process any @ImportResource annotations
AnnotationAttributes importResource =
	AnnotationConfigUtils.attributesFor(sourceClass.getMetadata(), ImportResource.class);
...
// Process individual @Bean methods
Set<MethodMetadata> beanMethods = retrieveBeanMethodMetadata(sourceClass);
for (MethodMetadata methodMetadata : beanMethods) {
	configClass.addBeanMethod(new BeanMethod(methodMetadata, configClass));
}
```

Reply:
What do you mean? That's what this PR already does.
It doesn't make sense to duplicate the `DataSourceProperties` bean creation, or the credentials setting IMO.